src/redux/csState.ts

```diff
@@ -49,6 +50,25 @@ export interface CsState {
   subscriptions: Subscriptions;
 }
 
+function updateValueCache(
+  oldValueCache: ValueCache,
+  newValueCache: ValueCache,
+  action: any
+): void {
+  const { pvName, value } = action.payload;
+  const pvState = oldValueCache[pvName];
+  let newValue: VType | undefined;
+  if (value instanceof VType) {
+    newValue = value;
+  } else {
+    newValue = mergeVtype(pvState.value, value);
+  }
+  const newPvState = Object.assign({}, pvState, {
+    value: newValue
+  });
+  newValueCache[action.payload.pvName] = newPvState;
+}
+
 export function csReducer(state = initialState, action: Action): CsState {
```

Comment:
What does `state = initialState` do?

Reply:
This is the way you initialise the store.

Reply:
Just wondering about the syntax - why is it not just `csReducer(initialState...`)? Does it take the value only the first time it is called or something like that? Or is it because initialState is the global state value we are trying to protect?